routes/packages.js

```diff
@@ -26,8 +29,19 @@ router.post('/packages', (req, res, next) => {
         res.status(422).json({ message: 'Could not get package from Shippo' });
       } else {
         Package.create(newPackage).then((item) => {
-          req.user.addPackage(item).then(() => {
-            res.json(item);
+          req.user.addPackage(item);
+          return item;
+        }).then((item) => {
+          PackageHistory.createFromShippoResponse(item.id, json).then((history) => {
+            res.json({
+              id: item.id,
+              carrier: item.carrier,
+              trackingNumber: item.trackingNumber,
+              history,
+            });
+          }).catch((err) => {
```

Comment:
I think this could be handled a bit better by just creating an error and throwing it up since we have an overarching catch. What do you think?